### ISSUE TITLE
Listview binded to ObservableCollection crashes on Android when selecting item. 

### ISSUE THREAD

**Issue by japero** [xamarin/xamarin-android#3198]
Description
Listview binded to ObservableCollection shows data correctly but when selecting item OnItemTapped event crashes app. I added break to event handler but it is never reach.

Here is the datasource

```
public ObservableCollection ZoneCompinedInfos
{
    get { return zoneCompinedInfos; }
    set
    {
        zoneCompinedInfos = value;
        this.OnPropertyChanged();
    }
}
```

And here is how i fill it when data is updated.

    ZoneCompinedInfos = new ObservableCollection(infos);

Of course I cannot show alldata due it is our app but this should give some idea?

Steps to Reproduce
Listview binded to ObservableCollection()
Listview shows data using DataTemplates to get binging to class properties.
Selecting item crashes app.
Expected Behavior
App shows View dispalying details about selected item

Actual Behavior
App crashes before hitting event handler break point.

Basic Information
Version with issue: Latest stable Xamarin version 4.0.0.482894
Last known good version: 3.5.0.274416 works just fine
IDE:
Platform Target Frameworks: tested on Android
iOS:
Android: 8.1
UWP:
Android Support Library Version:
Nuget Packages:
Affected Devices:
Screenshots
Reproduction Link
[error.txt](https://github.com/xamarin/xamarin-android/files/3276449/error.txt)

**Comment by brendanzagaeski** [comment]
Thanks for the report! Depending on which version of Visual Studio and the Xamarin.Android SDK you are using, I suspect you might be seeing <https://github.com/xamarin/xamarin-android/issues/3112>. To help confirm that, if you get a chance, if you could provide one of the following, that would be perfect:

* If you are using Visual Studio 2019 version 16.1, one option is to [follow the recent comment on #3112](https://github.com/xamarin/xamarin-android/issues/3112#issuecomment-500648563) to install the candidate development installer package that includes the candidate fix for that issue, and check if it resolves the issue you are seeing with your app.

* Or you can collect and attach back the `adb logcat` log from the crash:

    1. Open **Tools > Android > Android Adb Command Prompt**.
    2. Run the following command to clear the log on the device:

        ```cmd
        adb logcat -c
        ```

    3. Repeat the app crash once more.
    4. Run the following command to collect the log from the device:

        ```
        adb logcat -d > "%USERPROFILE%\Desktop\adblogcat.txt"
        ```

    5. Attach back the *adblogcat.txt* file from your desktop, or if you prefer, copy and paste just the section of it that starts with the "E/mono-rt" lines you included in the *error.txt* from the original description description and ends after the section that starts with a "Build fingerprint:" line and includes a "backtrace:" section (the native backtrace).

    The additional "backtrace:" information will often contain the name of the native function where the app exited unexpectedly, which can be helpful to distinguish among different native crashes.

Thanks in advance!

**Comment by japero** [comment]
Hi,

Sorry for late answer but attached is logcat exception. I cannot get this working no matter what I try. 
[android kaatuminen.txt](https://github.com/xamarin/xamarin-android/files/3319532/android.kaatuminen.txt)

**Comment by japero** [comment]
When I revert Xamarin nugets latest version to 3.5.0.274416 version problem dissappers!?
I just cannot get my head arounf this one what is the reason for error.

**Comment by brendanzagaeski** [comment]
Thanks for the additional log file, and apologies for the slow reply! Interestingly, Android is logging this error a little differently than for the other similar unmanaged faults I've seen recently. For some reason, this one doesn't have a backtrace. But this new log file shows that the issue also happens for you on an arm ABI application. That means this issue will have a different root cause than <https://github.com/xamarin/xamarin-android/issues/3112> because that issue only affects arm64 ABI applications.

```
06-20 13:23:46.670	Samsung SM-T719	Error	25389	DEBUG	signal 11 (SIGSEGV), code 2 (SEGV_ACCERR), fault addr 0xff224fe0
    r0 00000000  r1 bfadac60  r2 c15efa48  r3 bf814c68
    r4 00000000  r5 bfadac60  r6 00000000  r7 00000003
    r8 bf814c68  r9 ee705900  sl 00000000  fp ffa205d0
    ip ed10c000  sp ff224ff0  lr c4b2b770  pc cf0e0a80  cpsr 600f0010
06-20 13:23:46.670	Samsung SM-T719	Error	25389	DEBUG	pid: 25179, tid: 25179, name: l.mobileContact  >>> fi.semel.mobileContact <<<
06-20 13:23:46.670	Samsung SM-T719	Error	25389	DEBUG	ABI: 'arm'
06-20 13:23:46.669	Samsung SM-T719	Error	25389	DEBUG	Revision: '5'
06-20 13:23:46.669	Samsung SM-T719	Error	25389	DEBUG	Build fingerprint: 'samsung/gts28veltexx/gts28velte:7.0/NRD90M/T719XXU2BSA1:user/release-keys'
06-20 13:23:46.669	Samsung SM-T719	Error	25389	DEBUG	*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
06-20 13:23:46.580	Samsung SM-T719	Warning	390		debuggerd: handling request: pid=25179 uid=11524 gid=11524 tid=25179
06-20 13:23:46.579	Samsung SM-T719	Error	25179	libc	Fatal signal 11 (SIGSEGV), code 2, fault addr 0xff224fe0 in tid 25179 (l.mobileContact)
```

As one possible next step, if you might have a project you can zip up and share that demonstrates this problem, that would be excellent. If attaching the project to this GitHub issue publicly won't work, then a few other options are:

   * Put the example in a private GitHub repository, and add me (brendanzagaeski) as a collaborator.
   * Send a one to one message to me on Gitter, for example using a `/query @brendanzagaeski` command from within the [Xamarin.Android Gitter channel](https://gitter.im/xamarin/xamarin-android), and we can coordinate on another way to send the file non-publicly.
   * Or use **Help > Send Feedback > Report a Problem** from within Visual Studio to create an item on [Visual Studio Developer Community](https://developercommunity.visualstudio.com/spaces/8/index.html). Those allow private attachments.

Thanks in advance!

**Comment by japero** [comment]
Hi,

Long time since last message but I finally found out reason for this crash. 
I have a backgroud process taht receives messages and fires event that fills List with objects I created. 
I have made on some point Extender class for Ui to get IComparable and other stuff included and it "eats" this received object. Then on Ui I Binded labels etc. to this Extender class properties that provide only info from "master" object. Below is example.
Class is defined like this:
 public class ZoneCarInfoExtender:IEquatable<ZoneCarInfoExtender>,IComparable<ZoneCarInfoExtender>
    {
        public readonly ZoneCarInformation zoneCarInformation;

    public int Zone
        {
            get
            {
                    return zoneCarInformation.Zone;
            }
        }
....

After removed this extender class from use due I do not need IComparable etc. The error above disappeared and detail view is show!!!!
Hope this helps if someone want´s to find out reason for this behaviour.

Jarno
